src/meta/app/src/principal/user_grant.rs

```diff
@@ -267,7 +267,11 @@ impl UserGrantSet {
             .map(|e| {
                 if e.matches_entry(object) {
                     let mut e = e.clone();
-                    e.privileges ^= privileges;
+                    e.privileges = e
```

Comment:
Better add a test case?

Reply:
added in the sqlogictest part, i'll add a unit test about this